osquery/tables/system/windows/ntfs_acl_permissions.cpp

```diff
@@ -79,53 +117,18 @@ std::string accessPermsToStr(const unsigned long pmask) {
   return alg::join(permList, ",");
 }
 
-// helper function to get account/group name from trustee
-std::string trusteeToStr(const TRUSTEEW& trustee) {
-  // Max username length for Windows
-  const unsigned long maxBuffSize = 256;
+std::string pSidToStrUserName(PSID psid) {
   unsigned long sizeOut = maxBuffSize;
-  WCHAR name[maxBuffSize];
-  WCHAR domain[maxBuffSize];
+  char name[maxBuffSize];
+  char domain[maxBuffSize];
   SID_NAME_USE accountType;
-
-  switch (trustee.TrusteeForm) {
-  case TRUSTEE_IS_SID: {
-    // get the name from the SID
-    PSID psid = trustee.ptstrName;
-    auto r = LookupAccountSidW(
-        nullptr, psid, name, &sizeOut, domain, &sizeOut, &accountType);
-    if (r == FALSE) {
-      VLOG(1) << "LookupAccountSid error: " << GetLastError();
-      return "";
-    } else {
-      return wstringToString(name);
-    }
-  }
-  case TRUSTEE_IS_NAME:
-    // get the name from ptstrName
-    return wstringToString(trustee.ptstrName);
-  case TRUSTEE_BAD_FORM:
-    // Indicates a trustee form that is not valid.
-    // https://msdn.microsoft.com/en-us/library/windows/desktop/aa379638(v=vs.85).aspx
-    return "Invalid";
-  case TRUSTEE_IS_OBJECTS_AND_SID: {
-    // ptstrName member is a pointer to an OBJECTS_AND_SID struct
-    auto psid = reinterpret_cast<POBJECTS_AND_SID>(trustee.ptstrName)->pSid;
-    auto r = LookupAccountSidW(
-        nullptr, psid, name, &sizeOut, domain, &sizeOut, &accountType);
-    if (r == FALSE) {
-      VLOG(1) << "LookupAccountSid error: " << GetLastError();
-      return "";
-    } else {
-      return wstringToString(name);
-    }
-  }
-  case TRUSTEE_IS_OBJECTS_AND_NAME:
-    // ptstrName member is a pointer to an OBJECTS_AND_NAME struct
-    return wstringToString(
-        reinterpret_cast<OBJECTS_AND_NAME_W*>(trustee.ptstrName)->ptstrName);
-  default:
+  auto r = LookupAccountSidA(
```

Comment:
Is there a particular reason we are not using `LookupAccountSidA`?

Heads up @farfella, IIUC we should stick to using wide-string API versions.

Reply:
Yes. We should request changes if any -A calls are used.